rosbag2_storage_sqlite3/src/rosbag2_storage_sqlite3/sqlite_storage.cpp

```diff
@@ -323,9 +337,14 @@ void SqliteStorage::write(
 
   for (auto & message : messages) {
     write_locked(message);
+    // Update db_file_size_ with the estimated data size written to the DB
+    // + 3 * sizeof(int64_t) This is for storing timestamp, topic_id and index for messages table
+    db_file_size_ += message->serialized_data->buffer_length + 3 * sizeof(int64_t);
```

Comment:
Is this necessary?  We are going to throw all of this away after the `commit_transaction` below, so can we just skip this?

Reply:
Yes. It is required because we have two threads in rosbag2 recorder cache consumer and cache producer. There is possible a situation when we are dumping cache i.e. vector of messages to the disk by the cache consumer and at the same time cache producer requesting up-to-date `db_file_size` that will be needed on the upper layer to decide if bag already exceeds the limit and we need to make a bag split.